src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -21,9 +21,16 @@ public RegularExpressionAttribute(string pattern)
             : base(() => SR.RegexAttribute_ValidationError)
         {
             Pattern = pattern;
+            MatchTimeoutInMilliseconds = 2000;
         }
 
         /// <summary>
+        ///     Gets or sets the timeout to use when matching the regular expression pattern (in milliseconds)
+        ///     (-1 means never timeout).
+        /// </summary>
+        public int MatchTimeoutInMilliseconds { get; set; }
```

Comment:
You will need to add it to the contract reference assembly https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.Annotations/ref/System.ComponentModel.Annotations.cs and also bump the version number of the implementation and ref to 4.1.0.0.